nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -43,6 +63,25 @@ let
         '';
         };
 
+        hostname = mkOption {
+          type = with types; nullOr str;
+          description = "Set the hostname inside the container. Not required if using Host networking.";
```

Comment:
This implies that `network != "host" -> hostname != null`, but given that the option didn't previously exist, I'm guessing that it's fine to fall back to Docker's own defaults. Perhaps clarify this?

Reply:
Does 463120f resolve this?